qiskit/utils/deprecation.py

```diff
@@ -14,11 +14,26 @@
 
 import functools
 import warnings
-from typing import Type
+from typing import Any, Dict, Optional, Type
 
 
-def deprecate_arguments(kwarg_map, category: Type[Warning] = DeprecationWarning):
-    """Decorator to automatically alias deprecated argument names and warn upon use."""
+def deprecate_arguments(
+    kwarg_map: Dict[str, str],
+    category: Type[Warning] = DeprecationWarning,
+    *,
+    since: Optional[str] = None,  # pylint: disable=unused-argument
```

Comment:
```suggestion
    since: Optional[str] = None,
```

Better to `del` the unused argument to avoid getting out of sync.